dotnet/src/SemanticKernel.Abstractions/Diagnostics/HttpOperationException.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net;
+
+namespace Microsoft.SemanticKernel.Diagnostics;
+
+/// <summary>
+/// Represents an exception specific to HTTP operations.
+/// </summary>
+public class HttpOperationException : SKException
```

Comment:
I don't see much value in deriving it from SKException, as catching it as SKException ('catch(SKException ex)') is pointless from the consumer's perspective. This is because consumers cannot obtain failure details such as StatusCode, Response content, etc., without explicitly casting SKException to HttpOperationException. Most likely, consumer code developers will follow the standard practice of catching it explicitly as 'catch(HttpOperationException ex)' to access those details.

Regarding deriving it from HttpRequestException, the same train of thought applies in this case as well. It won't be possible to get those details without explicit casting of HttpRequestException to HttpOperationException. Additionally, HttpOperationException does not always represent HttpRequestException, as there are cases when it wraps the Azure.SDK RequestFailedException that the SDK throws if an HTTP operation fails.

Considering all of this, I will derive HttpOperationException from Exception for now until we have more insight to make an informed decision.